dashboard/signals.py

```diff
@@ -101,6 +75,8 @@ def handle_delete_certificate(sender, instance, **kwargs):  # pylint: disable=un
     """
     user = instance.user
     program = instance.course_run.course.program
-    program_enrollment = ProgramEnrollment.objects.filter(user=user, program=program).first()
-    if program_enrollment is not None:
+    try:
+        program_enrollment = ProgramEnrollment.objects.get(user=user, program=program)
         index_program_enrolled_users.delay([program_enrollment])
+    except ProgramEnrollment.DoesNotExist:
+        pass
```

Comment:
i'm not suggesting we change the try/except. i'm saying we move the whole block of code to a new helper method since it's repeated 4 times in this file 

```python
def fetch_and_index_enrollment(user, program):
    try:
        program_enrollment = ProgramEnrollment.objects.get(
            user=user,
            program=program
        )
        index_program_enrolled_users.delay([program_enrollment])
    except ProgramEnrollment.DoesNotExist:
        pass
```

on top of that, all 4 of them get `user` and `program` from `instance` in the exact same way. perhaps the method above could be added to search.tasks, and another helper method could be added here to extract `user` and `program` from `instance` and invoke it